[Automation] native-image run fails for io.netty:netty-codec-mqtt:4.1.79.Final

## native-image run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-mqtt:4.1.74.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25256364902/job/74056402160
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
oved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                        5.5s (7.1% of total time) in 768 GCs | Peak RSS: 1.73GB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests (executable)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library)
 /tmp/native-image-build-output.json (build_info)
========================================================================================================================
Finished generating 'io.netty.netty-codec-mqtt_tests-tests' in 1m 16s.
[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > invalidFactoryMessagesExposeDecoderFailure() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > payloadValueObjectsExposeUnmodifiableViewsAndReasonCodeLists() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > unsubscribeBuilderRoundTripsTopicFilters() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5ConnectMessageCarriesPropertiesAndBinaryCredentials() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > enumLookupsValidateProtocolLevelsMessageTypesAndReasonCodes() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > propertiesSupportSingleLookupMultipleUserPropertiesAndEmptySingleton() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations() FAILED


Failures (9):
  JUnit Jupiter:Netty_codec_mqttTest:mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation(Netty_codec_mqttTest.java:268)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       [...]
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
>      Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
Get more help at https://help.gradle.org.

BUILD FAILED in 1m 20s
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       java.base@25.0.3/java.lang.Class.ensureInitialized(DynamicHub.java:778)
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:unsubscribeBuilderRoundTripsTopicFilters()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'unsubscribeBuilderRoundTripsTopicFilters', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.unsubscribeBuilderRoundTripsTopicFilters(Netty_codec_mqttTest.java:378)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions(Netty_codec_mqttTest.java:314)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecodeAfterMqtt5Connect(Netty_codec_mqttTest.java:570)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation(Netty_codec_mqttTest.java:345)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads(Netty_codec_mqttTest.java:439)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip(Netty_codec_mqttTest.java:421)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat(Netty_codec_mqttTest.java:96)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations(Netty_codec_mqttTest.java:190)
       java.base@25.0.3/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.3/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.3/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.3/java.lang.Thread.runWith(Thread.java:1487)
       [...]

Test run finished after 8 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        14 tests found           ]
[         0 tests skipped         ]
[        14 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         9 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-mqtt:4.1.74.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 22s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-mqtt:4.1.74.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-mqtt:4.1.74.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25256364902/job/74056402162
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
nce.
------------------------------------------------------------------------------------------------------------------------
                       5.6s (6.7% of total time) in 769 GCs | Peak RSS: 1.62GiB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests (executable, 23.88MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'io.netty.netty-codec-mqtt_tests-tests' in 1m 22s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile

> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > invalidFactoryMessagesExposeDecoderFailure() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > payloadValueObjectsExposeUnmodifiableViewsAndReasonCodeLists() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > unsubscribeBuilderRoundTripsTopicFilters() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5ConnectMessageCarriesPropertiesAndBinaryCredentials() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > enumLookupsValidateProtocolLevelsMessageTypesAndReasonCodes() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > propertiesSupportSingleLookupMultipleUserPropertiesAndEmptySingleton() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations() FAILED


Failures (9):
  JUnit Jupiter:Netty_codec_mqttTest:mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation(Netty_codec_mqttTest.java:268)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:unsubscribeBuilderRoundTripsTopicFilters()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'unsubscribeBuilderRoundTripsTopicFilters', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.unsubscribeBuilderRoundTripsTopicFilters(Netty_codec_mqttTest.java:378)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions(Netty_codec_mqttTest.java:314)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecodeAfterMqtt5Connect(Netty_codec_mqttTest.java:570)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation(Netty_codec_mqttTest.java:345)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads(Netty_codec_mqttTest.java:439)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip(Netty_codec_mqttTest.java:421)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat(Netty_codec_mqttTest.java:96)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations(Netty_codec_mqttTest.java:190)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]

Test run finished after 7 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        14 tests found           ]
[         0 tests skipped         ]
[        14 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         9 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/reports/problems/problems-report.html
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 27s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-mqtt:4.1.74.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 28s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-mqtt:4.1.74.Final"]
```

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-mqtt:4.1.74.Final"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25256364902/job/74056402168
Last 300 lines of the log (excerpt truncated to fit the GitHub issue size limit):
```console
-----------------------------------------------------------------------------------------------
                       6.5s (7.8% of total time) in 761 GCs | Peak RSS: 1.79GiB | CPU load: 3.60
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests (executable, 27.44MiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /tmp/native-image-build-output.json (build_info, 1.04KiB)
========================================================================================================================
Finished generating 'io.netty.netty-codec-mqtt_tests-tests' in 1m 22s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
> Task :nativeTest FAILED
[junit-platform-native] WARNING: -djunit.platform.listeners.uid.tracking.output.dir not specified, trying to find test-ids on default Gradle/Maven locations. As this is a fallback mode, it could take a while. This should only happen if you are running tests executable manually and you didn't pass uid output directory with -djunit.platform.listeners.uid.tracking.output.dir=<path>.
[junit-platform-native] WARNING: Using test-ids from default Gradle project location:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/test-results/test/testlist
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/test-results/test/testlist] and its subfolders.
JUnit Platform on Native Image - report
----------------------------------------


WARNING: sun.misc.Unsafe::getLong has been called by io.netty.util.internal.PlatformDependent0$4 (file:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests)
WARNING: Please consider reporting this to the maintainers of class io.netty.util.internal.PlatformDependent0$4
WARNING: sun.misc.Unsafe::getLong will be removed in a future release
io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > invalidFactoryMessagesExposeDecoderFailure() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > payloadValueObjectsExposeUnmodifiableViewsAndReasonCodeLists() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > unsubscribeBuilderRoundTripsTopicFilters() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5ConnectMessageCarriesPropertiesAndBinaryCredentials() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > enumLookupsValidateProtocolLevelsMessageTypesAndReasonCodes() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > propertiesSupportSingleLookupMultipleUserPropertiesAndEmptySingleton() SUCCESSFUL

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat() FAILED

io_netty.netty_codec_mqtt.Netty_codec_mqttTest > publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations() FAILED


Failures (9):
  JUnit Jupiter:Netty_codec_mqttTest:mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation', methodParameterTypes = '']
    => java.lang.ExceptionInInitializerError
       io.netty.buffer.Unpooled.<clinit>(Unpooled.java:75)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.mqtt5PublishPropertiesRoundTripAfterProtocolVersionNegotiation(Netty_codec_mqttTest.java:268)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       [...]
     Caused by: java.lang.IllegalArgumentException: Can't find '[toLeakAwareBuffer]' in io.netty.buffer.AbstractByteBufAllocator
       io.netty.util.ResourceLeakDetector.addExclusions(ResourceLeakDetector.java:596)
       io.netty.buffer.AbstractByteBufAllocator.<clinit>(AbstractByteBufAllocator.java:37)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'pingDisconnectAndAuthMessagesUseFixedHeadersAndReasonCodeVariableHeaders', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.AbstractByteBufAllocator
       io.netty.buffer.ByteBufUtil.<clinit>(ByteBufUtil.java:86)
       io.netty.buffer.ByteBufAllocator.<clinit>(ByteBufAllocator.java:24)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:unsubscribeBuilderRoundTripsTopicFilters()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'unsubscribeBuilderRoundTripsTopicFilters', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.unsubscribeBuilderRoundTripsTopicFilters(Netty_codec_mqttTest.java:378)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.subscribeBuilderRoundTripsMultipleSubscriptionsAndSubscriptionOptions(Netty_codec_mqttTest.java:314)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecodeAfterMqtt5Connect(Netty_codec_mqttTest.java:570)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.mqtt5SubscribePropertiesAndSubscriptionOptionsRoundTripAfterProtocolVersionNegotiation(Netty_codec_mqttTest.java:345)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.messageFactoryCreatesTypedMessagesFromPublicHeadersAndPayloads(Netty_codec_mqttTest.java:439)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encodeAndDecode(Netty_codec_mqttTest.java:559)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.acknowledgementBuildersExposeReasonCodesPropertiesAndRoundTrip(Netty_codec_mqttTest.java:421)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.ByteBufAllocator
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:60)
       io.netty.channel.DefaultChannelConfig.<init>(DefaultChannelConfig.java:75)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:185)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:166)

       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:150)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:137)
       io.netty.channel.embedded.EmbeddedChannel.<init>(EmbeddedChannel.java:102)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.encode(Netty_codec_mqttTest.java:612)
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.connectBuilderSetsFlagsPayloadAndEncodesExpectedMqtt311WireFormat(Netty_codec_mqttTest.java:96)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       [...]
  JUnit Jupiter:Netty_codec_mqttTest:publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations()
    MethodSource [className = 'io_netty.netty_codec_mqtt.Netty_codec_mqttTest', methodName = 'publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations', methodParameterTypes = '']
    => java.lang.NoClassDefFoundError: Could not initialize class io.netty.buffer.Unpooled
       io_netty.netty_codec_mqtt.Netty_codec_mqttTest.publishMessageRoundTripsPayloadHeadersAndReferenceCountOperations(Netty_codec_mqttTest.java:190)
       java.base@25.0.2/java.lang.reflect.Method.invoke(Method.java:565)
       org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:727)
       org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
       org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
       org.junit.jupiter.api.AssertTimeoutPreemptively.lambda$submitTask$3(AssertTimeoutPreemptively.java:95)
       java.base@25.0.2/java.util.concurrent.FutureTask.run(FutureTask.java:328)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
       java.base@25.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
       java.base@25.0.2/java.lang.Thread.runWith(Thread.java:1487)
       [...]

Test run finished after 8 ms
[         3 containers found      ]
[         0 containers skipped    ]
[         3 containers started    ]
[         0 containers aborted    ]
[         3 containers successful ]
[         0 containers failed     ]
[        14 tests found           ]
[         0 tests skipped         ]
[        14 tests started         ]
[         0 tests aborted         ]
[         5 tests successful      ]
[         9 tests failed          ]


[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/reports/problems/problems-report.html
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/io.netty/netty-codec-mqtt/4.1.74.Final/build/native/nativeTestCompile/io.netty.netty-codec-mqtt_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 27s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
17 actionable tasks: 7 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for io.netty:netty-codec-mqtt:4.1.74.Final failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 28s
11 actionable tasks: 3 executed, 8 up-to-date
❌ ERROR: 'native-image run' for 4.1.79.Final failed with exit code 1.
FAILED[native-image run][4.1.79.Final][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="4.1.79.Final" ./gradlew clean test -Pcoordinates="io.netty:netty-codec-mqtt:4.1.74.Final"]
```

--- a/metadata/io.netty/netty-codec-mqtt/4.1.79.Final/reachability-metadata.json
+++ b/metadata/io.netty/netty-codec-mqtt/4.1.79.Final/reachability-metadata.json
@@ -25,39 +25,6 @@
         "typeReached": "io.netty.handler.codec.mqtt.MqttCodecUtil"
       },
       "type": "io.netty.util.DefaultAttributeMap$DefaultAttribute"
-    },
-    {
-      "condition": {
-        "typeReached": "io.netty.handler.codec.mqtt.MqttEncoder"
-      },
-      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueConsumerIndexField",
-      "fields": [
-        {
-          "name": "consumerIndex"
-        }
-      ]
-    },
-    {
-      "condition": {
-        "typeReached": "io.netty.handler.codec.mqtt.MqttEncoder"
-      },
-      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerIndexField",
-      "fields": [
-        {
-          "name": "producerIndex"
-        }
-      ]
-    },
-    {
-      "condition": {
-        "typeReached": "io.netty.handler.codec.mqtt.MqttEncoder"
-      },
-      "type": "io.netty.util.internal.shaded.org.jctools.queues.MpscArrayQueueProducerLimitField",
-      "fields": [
-        {
-          "name": "producerLimit"
-        }
-      ]
     }
   ]
 }

